### PR TITLE
Issue #24 Implement Gradle build for Java Spring application

### PR DESCRIPTION
**Pull request by jindong12**
Issue #24 

Applied & fixed changes in PR #228

### CODE REVIEW

File: C4SGSvc/src/main/resources/application.properties [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # Set here configurations for the database connection
 
 # Connection url for the database "netgloo_blog"
-spring.datasource.url = jdbc:mysql://localhost:3306/C4SG
+spring.datasource.url = jdbc:mysql://localhost:3306/c4sg
```
Comment (russon77) [inline C4SGSvc/src/main/resources/application.properties]: this change will force devs to create a new schema right? i ran into this error when i setup, where i had to create 'C4SG' and not 'c4sg'

Comment (jindong12) [inline C4SGSvc/src/main/resources/application.properties]: This change was made by PraniDahal @PranilDahal  and I just cherry pick this commit, and will change back later, thanks for pointing out.

File: C4SGSvc/build.gradle [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+repositories {
+        
+     maven { url "http://repo.maven.apache.org/maven2" }
+}
+dependencies {
+    compile(group: 'org.springframework.boot', name: 'spring-boot-starter-web', version:'1.4.2.RELEASE') {
+exclude(module: 'logback-classic')
+    }
+    compile group: 'org.springframework.boot', name: 'spring-boot-starter-jdbc', version:'1.4.2.RELEASE'
```
Comment (Oleggio) [inline C4SGSvc/build.gradle]: Should we move spring-boot, log4j, swagger general version numbers into property block?
E.g. in gradle-wrapper.properties or in build.gradle
`ext {
  springbootVersion:'1.4.2.RELEASE',
  ...
}
`
Then use like 
`compile group: 'org.springframework.boot', name: 'spring-boot-starter-data-jpa', version: ext.springbootVersion
`
etc

**Comment by jindong12** [comment]
Added more changes.

1. Enabled war building.
2. Add gradle wrapper (gradlew), to run gradle tasks without install

To run embed tomcat,

```
$ ./gradlew bootRun
```

To build war,

```
$ ./gradlew bootRepackage
```

**Pull request merged**